src/AspNet.Security.OAuth.Keycloak/KeycloakAuthenticationOptions.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+
+namespace AspNet.Security.OAuth.Keycloak
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="KeycloakAuthenticationHandler"/>.
+    /// </summary>
+    public class KeycloakAuthenticationOptions : OAuthOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KeycloakAuthenticationOptions"/> class.
+        /// </summary>
+        public KeycloakAuthenticationOptions()
+        {
+            ClaimsIssuer = KeycloakAuthenticationDefaults.Issuer;
+            CallbackPath = KeycloakAuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = KeycloakAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = KeycloakAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = KeycloakAuthenticationDefaults.UserInformationEndpoint;
+            Scope.Add("openid");
+            Scope.Add("email");
+            Scope.Add("roles");
+            ClaimActions.MapAll();
```

Comment:
MapAll() is required to map custom claims. others were the default scopes.

Reply:
I'm still not convinced this is needed. None of our other providers call it.

This feels like maybe something specific to your use case that would be configured in your call to `AddKeycloak()` when you configure the provider, rather than something it does by default.

Reply:
I followed the Okta options from here https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/97c1f79f4c09c492d10d1b7aec37aea2ceb49c64/src/AspNet.Security.OAuth.Okta/OktaAuthenticationOptions.cs#L27

but you are right you can add these from `AddKeycloak()` i have removed the default scopes.

Reply:
Is there a minimum scope required, or does the provider work out-of-the-box with exactly zero configured scopes?

The general idea is that the minimum scopes required to log in at all are configured in the provider, and any other additional scopes are configured by the user in their application.

My original point still stands though - none of the other providers need to call `MapAll()`, so I don't think this call should be here at all.

Reply:
yes, the minimum of `openid` is required for the login to work out of the box. I will update it and remove the `MapAll()`. It is needed if you have a custom claim in your app.